python/lancedb/table.py

```diff
@@ -697,6 +697,15 @@ def to_arrow(self) -> pa.Table:
         pa.Table"""
         return self._dataset.to_table()
 
+    def to_polars(self) -> "pl.DataFrame":
+        """Return the table as a polars DataFrame.
+
+        Returns
+        -------
+        pl.DataFrame
+        """
+        return pl.scan_pyarrow_dataset(self.to_lance())
```

Comment:
Should we expose `batch_size` as a parameter?  I think I'm ok with not doing so.  If users need fine grained control they can always call `scan_pyarrow_dataset` themselves.

Reply:
What are we referring to as `batch_size` here? It's a bit of confusing term - is that the same as polars [`set_streaming_chunk_size`](https://docs.pola.rs/py-polars/html/reference/api/polars.Config.set_streaming_chunk_size.html#polars.Config.set_streaming_chunk_size)? It's good to drop a description here as to what we mean explicitly.

Reply:
<img width="913" alt="image" src="https://github.com/lancedb/lancedb/assets/759245/19b74cfc-e9da-4835-86f8-d54b600e1b4c">

Reply:
That description for `batch_size` makes sense. The type annotation for `LazyFrame` also makes more sense now!